fca_import_scripts/lib/ext_to_sql.rb

```diff
@@ -0,0 +1,133 @@
+require 'csv'
+require 'date'
+require 'yaml'
+
+class ExtToSql
+  TIMESTAMP = DateTime.now.strftime('%Y-%m-%d %H:%M:%S.%N')
+  REPAIR_FILE = File.join(File.dirname(__FILE__), '..', 'repairs.yml')
+
+  def initialize(stderr = nil)
+    @stderr = stderr
+  end
+
+  def process_ext_file(path, &block)
+    File.foreach(path) do |line|
+      line = repair_line line
+      row = line.split('|')
+
+      next if row[0] == 'Footer'
+
+      if row[0] == 'Header'
+        @type = determine_type_from_header(row)
+        block.call build_copy_statement_for_type(@type)
+        log "  • \033[33;36mConverting #{@type} EXT to SQL.\033[0m ", newline: false
+        next
+      end
+
+      if record_active?(@type, row)
+        block.call build_row(row)
+      end
+
+      write_progress
+    end
+
+    block.call end_copy_statement
+  end
+
+  private
+
+  def determine_type_from_header(row)
+    case row[1]
+    when 'Alternative Firm Name'
+      return :subsidiary
```

Comment:
not being too familiar with the case syntax, are the explicit return statements necessary?